arkane/ess/qchem.py

```diff
@@ -57,8 +57,64 @@ class QChemLog(ESSAdapter):
     arrays. QChemLog is an adapter for the abstract class ESSAdapter.
     """
 
-    def __init__(self, path):
+    def __init__(self, path, check_for_errors=True):
         self.path = path
+        if check_for_errors:
+            self.check_for_errors()
+
+    def check_for_errors(self):
+        """
+        Checks for common errors in a QChem log file.
+        If any are found, this method will raise an error and crash.
+        """
+        with open(os.path.join(self.path), 'r') as f:
+            lines = f.readlines()
+            error, warning_line, warning, warn_message = None, None, None, None
+            for line in reversed(lines):
+                # check for common error messages
+                if 'SCF failed' in line:
+                    error = 'SCF failed'
+                    break
+                elif 'error' in line and 'DIIS' not in line and 'gprntSymmMtrx' not in line \
+                        and 'Relative error' not in line and 'zonesort' not in line:
+                    # these are **normal** lines that we should not capture:
+                    # "SCF converges when DIIS error is below 1.0E-08", or
+                    # "Cycle       Energy         DIIS Error" or
+                    # "gprntSymmMtrx error report: "
+
+                    # "Relative error" is captured as a warning later
+
+                    # If running on NERSC, also want to avoid lines containing 'zonesort'
+                    # Per the NERSC documenation:
+                    # The error message means that running the zonesort kernel failed for some reason.
+                    # The end result is that your code may have run less optimally.
+                    # Other than that, the message is usually harmless.
+                    # https://docs.nersc.gov/jobs/errors/
+                    error = 'SCF failed'
```

Comment:
Are you sure that this error will always be an SCF error?

Reply:
To my knowledge, I think so? I only have a few examples of QChem failing for me so perhaps there are exceptions. But this builds upon the logic in ARC's [troubleshooting](https://github.com/ReactionMechanismGenerator/ARC/blob/master/arc/job/trsh.py#L172-L207)

Reply:
Let's match ARC for now then